<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>SARS Genomic Surveillance</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../materials/03-case_studies/03-eqa.html" rel="next">
<link href="../../materials/03-case_studies/01-switzerland.html" rel="prev">
<link href="../../site_libs/quarto-contrib/quarto-project/cambiotraining/courseformat/img/university-of-cambridge-favicon.ico" rel="icon">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../site_libs/quarto-contrib/quarto-project/cambiotraining/courseformat/img/university_crest_reversed.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Bioinformatics for SARS Genomic Surveillance</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/cambiotraining/sars-cov-2-genomics" rel="" target=""><i class="bi bi-github" role="img" aria-label="Bioinformatics Training Facility GitHub">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/bioinfocambs" rel="" target=""><i class="bi bi-twitter" role="img" aria-label="Bioinformatics Training Facility Twitter">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://genomic.social/@BioInfoCambs" rel="" target=""><i class="bi bi-mastodon" role="img" aria-label="Bioinformatics Training Facility Mastodon">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../materials/03-case_studies/01-switzerland.html">Case Studies</a></li><li class="breadcrumb-item"><a href="../../materials/03-case_studies/02-southafrica.html"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">South Africa (Illumina)</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Welcome</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../setup.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data &amp; Software</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Introduction</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/01-intro/01-surveillance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">SARS-CoV-2 genomic surveillance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/01-intro/02-ngs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction to NGS</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/01-intro/03-workflows.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Bioinformatic workflows</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Clinical Isolates</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/02-isolates/01-consensus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Consensus assembly</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/02-isolates/02-qc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Quality Control</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/02-isolates/03-lineages.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Lineages and variants</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/02-isolates/04-phylogeny.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Building phylogenetic trees</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Case Studies</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/03-case_studies/01-switzerland.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Switzerland (Nanopore)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/03-case_studies/02-southafrica.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">South Africa (Illumina)</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/03-case_studies/03-eqa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">EQA (Exercise)</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">Wastewater Samples</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/04-wastewater/01-ww_surveillance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Wastewater Surveillance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/04-wastewater/02-ww_abundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Lineage Abundance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/04-wastewater/03-ww_visualisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Abundance Visualisation</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/04-wastewater/04-ww_mutations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Mutation Analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true">
 <span class="menu-text">Software</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/05-software/01-managing_software.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Managing Bioinformatics Software</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/05-software/03-software_setup.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Software setup</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/appendices/file_formats.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Common file formats</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/appendices/unix_cheatsheet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Unix cheat sheet</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/appendices/quick_r.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R fundamentals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../materials/appendices/tools_and_resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tools and Resources</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#pipeline-overview" id="toc-pipeline-overview" class="nav-link active" data-scroll-target="#pipeline-overview"><span class="header-section-number">9.1</span> Pipeline Overview</a></li>
  <li><a href="#preparing-files" id="toc-preparing-files" class="nav-link" data-scroll-target="#preparing-files"><span class="header-section-number">9.2</span> Preparing Files</a>
  <ul class="collapse">
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">9.2.1</span> Data</a></li>
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata"><span class="header-section-number">9.2.2</span> Metadata</a></li>
  </ul></li>
  <li><a href="#consensus-assembly" id="toc-consensus-assembly" class="nav-link" data-scroll-target="#consensus-assembly"><span class="header-section-number">9.3</span> Consensus Assembly</a>
  <ul class="collapse">
  <li><a href="#samplesheet" id="toc-samplesheet" class="nav-link" data-scroll-target="#samplesheet"><span class="header-section-number">9.3.1</span> Samplesheet</a></li>
  <li><a href="#running-viralrecon" id="toc-running-viralrecon" class="nav-link" data-scroll-target="#running-viralrecon"><span class="header-section-number">9.3.2</span> Running Viralrecon</a></li>
  </ul></li>
  <li><a href="#consensus-quality" id="toc-consensus-quality" class="nav-link" data-scroll-target="#consensus-quality"><span class="header-section-number">9.4</span> Consensus Quality</a>
  <ul class="collapse">
  <li><a href="#general-metrics" id="toc-general-metrics" class="nav-link" data-scroll-target="#general-metrics"><span class="header-section-number">9.4.1</span> General Metrics</a></li>
  <li><a href="#variants" id="toc-variants" class="nav-link" data-scroll-target="#variants"><span class="header-section-number">9.4.2</span> Variants</a></li>
  </ul></li>
  <li><a href="#clean-fasta" id="toc-clean-fasta" class="nav-link" data-scroll-target="#clean-fasta"><span class="header-section-number">9.5</span> Clean FASTA</a>
  <ul class="collapse">
  <li><a href="#missing-intervals" id="toc-missing-intervals" class="nav-link" data-scroll-target="#missing-intervals"><span class="header-section-number">9.5.1</span> Missing Intervals</a></li>
  </ul></li>
  <li><a href="#downstream-analyses" id="toc-downstream-analyses" class="nav-link" data-scroll-target="#downstream-analyses"><span class="header-section-number">9.6</span> Downstream Analyses</a>
  <ul class="collapse">
  <li><a href="#lineage-assignment" id="toc-lineage-assignment" class="nav-link" data-scroll-target="#lineage-assignment"><span class="header-section-number">9.6.1</span> Lineage Assignment</a></li>
  <li><a href="#phylogeny" id="toc-phylogeny" class="nav-link" data-scroll-target="#phylogeny"><span class="header-section-number">9.6.2</span> Phylogeny</a></li>
  <li><a href="#clustering" id="toc-clustering" class="nav-link" data-scroll-target="#clustering"><span class="header-section-number">9.6.3</span> Clustering</a></li>
  </ul></li>
  <li><a href="#integration-visualisation" id="toc-integration-visualisation" class="nav-link" data-scroll-target="#integration-visualisation"><span class="header-section-number">9.7</span> Integration &amp; Visualisation</a></li>
  <li><a href="#bonus-full-workflow" id="toc-bonus-full-workflow" class="nav-link" data-scroll-target="#bonus-full-workflow"><span class="header-section-number">9.8</span> Bonus: Full Workflow</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">South Africa (Illumina)</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Learning Objectives
</div>
</div>
<div class="callout-body-container callout-body">
<p>This section demonstrates a start-to-finish analysis of a dataset sequenced on an <em>Illumina</em> platform, using the concepts and tools covered in previous sections. You can download the data from these links (two versions available):</p>
<ul>
<li><a href="https://www.dropbox.com/sh/6jnpv7ui2xm8f9h/AAC0MRuPfSsNmBlCNvI-wiKna?dl=1">South Africa Case Study - Full Data</a> – this includes data for 24 samples, which gives a more realistic sample size, but can take a few hours to run on a small computer.</li>
<li><a href="https://www.dropbox.com/sh/2u0svuq7rre8261/AACisGtF21x3Ou7Xrj_W4DaRa?dl=1">South Africa Case Study - Small Version</a> – this includes data for a subset of 8 samples, which is more suitable for training purposes (but the results will look slightly different from the ones shown here).</li>
</ul>
<p>By the end of this section, you should be able to:</p>
<ul>
<li>Prepare all the files necessary to run the consensus pipeline.</li>
<li>Run the <em>viralrecon</em> pipeline to generate FASTA consensus from raw FASTQ files.</li>
<li>Assess and collect several quality metrics for the consensus sequences.</li>
<li>Clean output files, in preparation for other downstream analysis.</li>
<li>Assign sequences to lineages using <em>Nextclade</em> and/or <em>Pangolin</em>.</li>
<li>Contextualise your sequences in other background data and cluster them based on phylogenetic analysis.</li>
<li>Integrate the metadata and results to generate useful visualisations of your data.</li>
<li>Report your analysis.</li>
</ul>
</div>
</div>
<p>We will analyse data from 24 samples collected in South Africa between Nov-Dec 2021. The samples were sequenced on an <em>Illumina MiSeq</em> platform.</p>
<p>The final product of our work (and main objective) is to produce a report of the analysis, which you can see here: <a href="https://docs.google.com/document/d/1yisycYBj-rmSGzTOzA5_ELfNYP1NKHHWj1rQoHqKGw8/edit?usp=sharing" target="_blank">South Africa Case Study Report</a>.</p>
<p>In summary, the report addresses the following:</p>
<ul>
<li>What was the quality of the consensus sequences?</li>
<li>What lineage/clade was each of our samples assigned to?</li>
<li>How many clusters of samples did we identify?</li>
<li>How did the detected lineages change over the time of sampling?</li>
</ul>
<p>We also produce several essential output files, which would usually be necessary to upload our data to public repositories:</p>
<ul>
<li>Metadata (CSV)</li>
<li>Consensus sequences (FASTA)</li>
<li>Consensus sequence quality metrics (CSV)</li>
<li>Variants (CSV)</li>
</ul>
<section id="pipeline-overview" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="pipeline-overview"><span class="header-section-number">9.1</span> Pipeline Overview</h2>
<p>Our analysis starts with <strong>FASTQ files</strong>, which will be used with the <strong><code>nf-core/viralrecon</code> <em>Nextflow</em> pipeline</strong>. This will give us several <strong>quality control</strong> metrics essential for our downstream analysis and reporting.</p>
<p>Critical files output by the pipeline will need to be further processed, including combining our <strong>consensus FASTA files</strong> and obtaining a list of <strong>filtered SNP/indel variants</strong>. Using these clean files, we can then proceed to downstream analysis, which includes assigning each sample to the most up-to-date <strong>Pango lineage</strong>, <strong>Nextclade clade</strong> and <strong>WHO designation</strong>. Finally, we can do more advanced analysis, including the idenfication of <strong>sample clusters</strong> based on phylogenetic analysis, or produce timeseries visualisations of mutations or variants of concern. With all this information together, we will have the necessary pieces to submit our results to <strong>public repositories</strong> and write <strong>reports</strong> to inform public health decisions.</p>
<p><img src="images/analysis_overview.png" class="img-fluid"></p>
</section>
<section id="preparing-files" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="preparing-files"><span class="header-section-number">9.2</span> Preparing Files</h2>
<p>Before we start our work, it’s always a good idea to setup our directory structure, so we keep our files organised as the analysis progresses. From the data we are starting with, we already have the following directories:</p>
<ul>
<li><code>data</code> → contains the sequencing data in a sub-directory called <code>reads</code>.</li>
<li><code>resources</code> → files that were downloaded from public repositories.</li>
<li><code>scripts</code> → <em>bash</em> and <em>R</em> scripts used to run the analysis.</li>
</ul>
<p>We create two additional directories:</p>
<ul>
<li><code>report</code> → files and documents that we report to our colleagues or upload to public repositories.</li>
<li><code>results</code> → results of the analysis.</li>
</ul>
<p>You can create directories from the command line using the <code>mkdir</code> command:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> results</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> report</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="data" class="level3" data-number="9.2.1">
<h3 data-number="9.2.1" class="anchored" data-anchor-id="data"><span class="header-section-number">9.2.1</span> Data</h3>
<p>We start our analysis from FASTQ files generated by the <em>Illumina</em> sequencer. As this is paired-end sequencing, we have two files per sample (with suffix <code>_1</code> and <code>_2</code>):</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> data/reads</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>SRR17051908_1.fastq.gz  SRR17051953_1.fastq.gz  SRR17461700_1.fastq.gz  SRR17712594_1.fastq.gz
SRR17051908_2.fastq.gz  SRR17051953_2.fastq.gz  SRR17461700_2.fastq.gz  SRR17712594_2.fastq.gz
SRR17051916_1.fastq.gz  SRR17054503_1.fastq.gz  SRR17461712_1.fastq.gz  SRR17712607_1.fastq.gz
SRR17051916_2.fastq.gz  SRR17054503_2.fastq.gz  SRR17461712_2.fastq.gz  SRR17712607_2.fastq.gz
SRR17051923_1.fastq.gz  SRR17088917_1.fastq.gz  SRR17701832_1.fastq.gz  SRR17712711_1.fastq.gz
SRR17051923_2.fastq.gz  SRR17088917_2.fastq.gz  SRR17701832_2.fastq.gz  SRR17712711_2.fastq.gz
SRR17051932_1.fastq.gz  SRR17088924_1.fastq.gz  SRR17701841_1.fastq.gz  SRR17712779_1.fastq.gz
SRR17051932_2.fastq.gz  SRR17088924_2.fastq.gz  SRR17701841_2.fastq.gz  SRR17712779_2.fastq.gz
SRR17051935_1.fastq.gz  SRR17088928_1.fastq.gz  SRR17701890_1.fastq.gz  SRR17712994_1.fastq.gz
SRR17051935_2.fastq.gz  SRR17088928_2.fastq.gz  SRR17701890_2.fastq.gz  SRR17712994_2.fastq.gz
SRR17051951_1.fastq.gz  SRR17088930_1.fastq.gz  SRR17712442_1.fastq.gz  SRR17712997_1.fastq.gz
SRR17051951_2.fastq.gz  SRR17088930_2.fastq.gz  SRR17712442_2.fastq.gz  SRR17712997_2.fastq.gz</code></pre>
</section>
<section id="metadata" class="level3" data-number="9.2.2">
<h3 data-number="9.2.2" class="anchored" data-anchor-id="metadata"><span class="header-section-number">9.2.2</span> Metadata</h3>
<p>Metadata for these samples is available in the file <code>sample_info.csv</code>. Here is some of the information we have available for these samples:</p>
<ul>
<li><code>sample</code> → the sample ID.</li>
<li><code>collection_date</code> → the date of collection for the sample in the format YYYY-MM-DD.</li>
<li><code>country</code> → the country of origin for this sample.</li>
<li><code>geo_loc_region</code> → the region within the country where the sample was collected.</li>
<li><code>latitude</code>/<code>longitude</code> → coordinates for sample location (in this case we’re only given a single coordinate for the whole country - in a real setting you may want to collect a precise location).</li>
<li><code>sequencing_instrument</code> → the model for the sequencing instrument used (e.g.&nbsp;NovaSeq 6000, MinION, etc.).</li>
<li><code>sequencing_protocol_name</code> → the type of protocol used to prepare the samples (e.g.&nbsp;ARTIC).</li>
<li><code>amplicon_primer_scheme</code> → for amplicon protocols, what version of the primers was used (e.g.&nbsp;V3, V4.1)</li>
</ul>
</section>
</section>
<section id="consensus-assembly" class="level2" data-number="9.3">
<h2 data-number="9.3" class="anchored" data-anchor-id="consensus-assembly"><span class="header-section-number">9.3</span> Consensus Assembly</h2>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>See <a href="../02-isolates/01-consensus.html#sec-consensus"><span>Section&nbsp;4.1</span></a>, if you need to revise how the <code>nf-core/viralrecon</code> pipeline works.</p>
</div>
</div>
<p>The first step in the bioinformatic analysis is to run the <code>nf-core/viralrecon</code> pipeline. But first we need to prepare our input files.</p>
<section id="samplesheet" class="level3" data-number="9.3.1">
<h3 data-number="9.3.1" class="anchored" data-anchor-id="samplesheet"><span class="header-section-number">9.3.1</span> Samplesheet</h3>
<p>For <em>Illumina</em> data, we need a <strong>samplesheet CSV file</strong> with three columns, indicating sample name (first column) and the respective FASTQ file paths for read 1 (second column) and read 2 (third column).</p>
<p>Because our FASTQ file names are not very user-friendly, we used some command-line tricks to help us produce this table:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># list read 1 files and save output in a temporary file</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> data/reads/<span class="pp">*</span>_1.fastq.gz <span class="op">&gt;</span> read1_filenames.txt</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># list read 2 files and save output in a temporary file</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> data/reads/<span class="pp">*</span>_2.fastq.gz <span class="op">&gt;</span> read2_filenames.txt</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># initiate a file with column names</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"fastq_1,fastq_2"</span> <span class="op">&gt;</span> samplesheet.csv</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co"># paste the two temporary files together, using comma as a delimiter</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span> <span class="at">-d</span> <span class="st">","</span> read1_filenames.txt read2_filenames.txt <span class="op">&gt;&gt;</span> samplesheet.csv</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the two temporary files</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> read1_filenames.txt read2_filenames.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>These commands resulted in creating a file called <code>samplesheet.csv</code>, which contains the following:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>fastq_1,fastq_2
data/reads/SRR17051908_1.fastq.gz,data/reads/SRR17051908_2.fastq.gz
data/reads/SRR17051916_1.fastq.gz,data/reads/SRR17051916_2.fastq.gz
data/reads/SRR17051923_1.fastq.gz,data/reads/SRR17051923_2.fastq.gz
data/reads/SRR17051932_1.fastq.gz,data/reads/SRR17051932_2.fastq.gz
data/reads/SRR17051935_1.fastq.gz,data/reads/SRR17051935_2.fastq.gz
data/reads/SRR17051951_1.fastq.gz,data/reads/SRR17051951_2.fastq.gz
data/reads/SRR17051953_1.fastq.gz,data/reads/SRR17051953_2.fastq.gz
data/reads/SRR17054503_1.fastq.gz,data/reads/SRR17054503_2.fastq.gz
data/reads/SRR17088917_1.fastq.gz,data/reads/SRR17088917_2.fastq.gz</code></pre>
<p>So, we programmatically created the last two columns of our file. We then opened this CSV file in <em>Excel</em> to add another column “sample” where we included our sample names and saved the file again as a CSV format. Here are the top few rows of the final file:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>sample,fastq_1,fastq_2
ZA01,data/reads/SRR17051908_1.fastq.gz,data/reads/SRR17051908_2.fastq.gz
ZA02,data/reads/SRR17051923_1.fastq.gz,data/reads/SRR17051923_2.fastq.gz
ZA03,data/reads/SRR17051916_1.fastq.gz,data/reads/SRR17051916_2.fastq.gz
ZA04,data/reads/SRR17051953_1.fastq.gz,data/reads/SRR17051953_2.fastq.gz
ZA05,data/reads/SRR17051951_1.fastq.gz,data/reads/SRR17051951_2.fastq.gz
ZA06,data/reads/SRR17051935_1.fastq.gz,data/reads/SRR17051935_2.fastq.gz
ZA07,data/reads/SRR17051932_1.fastq.gz,data/reads/SRR17051932_2.fastq.gz
ZA08,data/reads/SRR17054503_1.fastq.gz,data/reads/SRR17054503_2.fastq.gz
ZA09,data/reads/SRR17088930_1.fastq.gz,data/reads/SRR17088930_2.fastq.gz</code></pre>
</section>
<section id="running-viralrecon" class="level3" data-number="9.3.2">
<h3 data-number="9.3.2" class="anchored" data-anchor-id="running-viralrecon"><span class="header-section-number">9.3.2</span> Running Viralrecon</h3>
<p>Now we are ready to run the <code>nf-core/viralrecon</code> pipeline (see <a href="../02-isolates/01-consensus.html#sec-viralrecon"><span>Section&nbsp;4.3</span></a> for details). We saved our command in a script (<code>scripts/01-run_viralrecon.sh</code>), which we created with the command line text editor <code>nano</code>. This ensures that our analysis is <strong>reproducible</strong> and <strong>traceable</strong> (we can go back to the script to see how the analysis was run).</p>
<p>First, we activate our software environment, to ensure <em>Nextflow</em> is available to us:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate nextflow</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Then, we run our script with:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> scripts/01-run_viralrecon.sh</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Which will start executing the pipeline.</p>
<p>For reference, here is the command included in that script:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nextflow</span> run nf-core/viralrecon <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">-r</span> 2.6.0 <span class="at">-profile</span> singularity <span class="dt">\</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--max_memory</span> <span class="st">'15.GB'</span> <span class="at">--max_cpus</span> 8 <span class="dt">\</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--platform</span> illumina <span class="dt">\</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input</span> samplesheet.csv <span class="dt">\</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--outdir</span> results/viralrecon <span class="dt">\</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--protocol</span> amplicon <span class="dt">\</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--genome</span> <span class="st">'MN908947.3'</span> <span class="dt">\</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">--primer_set</span> artic <span class="dt">\</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">--primer_set_version</span> 3 <span class="dt">\</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">--skip_assembly</span> <span class="at">--skip_asciigenome</span> <span class="dt">\</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">--skip_pangolin</span> <span class="at">--skip_nextclade</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
The <code>\</code> in long commands
</div>
</div>
<div class="callout-body-container callout-body">
<p>In the command above, you will notice several <code>\</code> at the end of each line. This is indicating that we want to continue writing our command in the next line. Notice that the last line <em>does not</em> include <code>\</code>, because that is the end of the command. This is very useful when commands are very long, because it makes the code more readable.</p>
</div>
</div>
<p>After running the pipeline, we notice the following message:</p>
<pre><code>-[nf-core/viralrecon] 2 samples skipped since they failed Bowtie2 1000 mapped read threshold:
    201: ZA09
    176: ZA12</code></pre>
<p>This indicates that two samples - ZA09 and ZA12 - had very few reads aligned to the SARS-CoV-2 genome. Could the reason be that they had very few reads to start with? We can quickly investigate this hypothesis by counting the number of lines in the FASTQ files from these samples:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># FASTQ file for ZA09</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">zcat</span> data/reads/SRR17088930_1.fastq.gz <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>698044</code></pre>
<p>Since each sequence takes 4 lines in a FASTQ format, this indicates that this sample had 698044/4 = 124511 reads. That’s certainly more than 1000 reads, so the reason the sample failed must be something else. We will investigate this further in the next section.</p>
</section>
</section>
<section id="consensus-quality" class="level2" data-number="9.4">
<h2 data-number="9.4" class="anchored" data-anchor-id="consensus-quality"><span class="header-section-number">9.4</span> Consensus Quality</h2>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>See <a href="../02-isolates/02-qc.html#sec-consensus-qc"><span>Section&nbsp;5.2</span></a>, if you need to revise how to assess the quality of consensus sequences.</p>
</div>
</div>
<section id="general-metrics" class="level3" data-number="9.4.1">
<h3 data-number="9.4.1" class="anchored" data-anchor-id="general-metrics"><span class="header-section-number">9.4.1</span> General Metrics</h3>
<p>We used the <em>MultiqQC</em> report to assess the initial quality of our samples. The quality report can be found in <code>results/viralrecon/multiqc/multiqc_report.html</code>.</p>
<p>We paid particular attention to:</p>
<ul>
<li>Number of reads mapped to the reference genome.</li>
<li>Median depth of coverage.</li>
<li>Percentage of the genome with missing bases (‘N’).</li>
<li>Number of SNP + Indel variants.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/southafrica_multiqc_metrics.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Example <em>MultiQC</em> output, highlighting the main columns with quality information that will be collected for the final report. Note that some columns have been ommited here for clarity.</figcaption>
</figure>
</div>
<p>We noted that:</p>
<ul>
<li>2 samples - ZA09 and ZA12 - completely failed. This was already noted after we ran the pipeline, above. The reason seems to be because these samples have a very low percentage of non-human reads (less than 2%), indicating that the sequenced material was mostly human. Because of this, there were not enough reads to assemble a genome.</li>
<li>2 other samples - ZA10 and ZA23 - had &gt; 90% missing bases, also indicating a very poor assembly. The reason was the same as above, both samples had low % of non-human reads.</li>
<li>1 samples - ZA14 - had ~39% missing bases. In this case the reason was not a low number of mapped reads. For example, sample ZA13 had a very similar median depth of coverage (121 vs 161 in ZA14) and even fewer mapped reads (24k vs 60k in ZA14). But ZA13 only had 2% missing bases. However, upon inspection of the amplicon heatmap, we detected that several amplicons were not properly amplified in ZA14 compared to ZA13. Therefore, the reason for high % missing bases in ZA14 was due to low amplification efficiency in this sample.</li>
<li>There was some systematic dropout for some amplicons, in particular <code>nCoV-2019_64</code> had very low amplification in several of the samples. Of note was also <code>nCoV-2019_73</code>, and other neighbouring amplicons.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/southafrica_amplicon_heatmap.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Amplicon depth of sequencing (or coverage) across samples, from the <em>MultiQC</em> report. Darker regions indicate systematic amplicon dropout, probably due to issues during the PCR amplification.</figcaption>
</figure>
</div>
<p>Besides the <em>MultiQC</em> report, the pipeline also outputs a CSV file with collected summary metrics (equivalent to the first table on the report): <code>results/viralrecon/multiqc/summary_variants_metrics_mqc.csv</code>. We will use this file later to join this information with our metadata and lineage assignment using the <em>R</em> software (detailed in “Integration &amp; Visualisation” section, below).</p>
</section>
<section id="variants" class="level3" data-number="9.4.2">
<h3 data-number="9.4.2" class="anchored" data-anchor-id="variants"><span class="header-section-number">9.4.2</span> Variants</h3>
<p>We also looked at the table of variants obtained from the pipeline. This is output in <code>results/viralrecon/variants/ivar/variants_long_table.csv</code>. This table can be very useful to keep track of particular mutations that may be increasing over time. Later, we will tidy this table to attach to our reported results (“Integration &amp; Visualisation” section).</p>
<p>But for now, we will explore this table to address a few more quality-related questions. We opened this table in <em>Excel</em> to answer the following:</p>
<ul>
<li>Where there samples with a high number of intermediate allele frequencies? This could indicate mixed samples due to cross-contamination.<br>
</li>
<li>Where there samples with frameshift mutations? These mutations should be rare because they are highly disruptive to the functioning of the virus. So, their occurrence may be due to errors rather than a true mutation and it’s good to make a note of this.</li>
</ul>
<p>By manual inspection of this table (and using the “filter” feature in <em>Excel</em>), we found 74 variants with alternative allele frequency less than 75%. These were spread across samples, all samples having less than 10 such low-frequency mutations, and most samples having less than 5. Sample ZA23 - which we had previously highlighted has having a high % missing bases (39%) - had 8 low-frequency mutations out of a total of 48 mutations in this sample (~16%), suggesting further quality issues in this sample.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/switzerland_variants.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Variants table output by <em>viralrecon</em>, with a summary of the main columns of interest. Note that this table also includes a column with lineage assignment (not shown in this snapshot). Remember that at this stage we mostly ignore this column, as <em>viralrecon</em> does not use the most up-to-date version of the lineage databases.</figcaption>
</figure>
</div>
</section>
</section>
<section id="clean-fasta" class="level2" data-number="9.5">
<h2 data-number="9.5" class="anchored" data-anchor-id="clean-fasta"><span class="header-section-number">9.5</span> Clean FASTA</h2>
<p>The pipeline outputs the consensus sequences in <code>results/viralrecon/variants/ivar/consensus/bcftools/*.consensus.fa</code> (one file for each sample). For downstream analysis, it is convenient to combine all these sequences into a single file, and also clean the sequence names (to remove some text – ” MN908947.3” –, which is added by the <code>bcftools</code> variant caller).</p>
<p>We created a new script to clean our consensus FASTA files, which we ran with <code>bash scripts/02-clean_fasta.sh</code>:</p>
<!-- 
TODO: possibly we don't want to include the low coverage samples ZA23 
This might be convoluted, but works if we just want to use code:

```
cat sample_info.csv | grep -v "ZA09" | grep -v "ZA12" | grep -v "ZA10" | cut -d , -f 1 > ids.txt
cat results/viralrecon/medaka/*.consensus.fasta | sed 's| MN908947.3||' | seqkit grep -f ids.txt --by-name > report/consensus.fa
```
-->
<div class="sourceCode" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># combine and clean FASTA files</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> results/viralrecon/variants/ivar/consensus/bcftools/<span class="pp">*</span>.consensus.fa <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s| MN908947.3||'</span> <span class="op">&gt;</span> report/consensus.fa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This command does two things:</p>
<ul>
<li>Combine all our FASTA consensus sequences into a single file (using <code>cat</code>).</li>
<li>Clean the sequence names (using <code>sed</code>).</li>
</ul>
<p>The output was saved as a new FASTA file: <code>report/consensus.fa</code>.</p>
<!-- 
Another way is using `seqkit`: 
`cat results/viralrecon/medaka/*.consensus.fasta | seqkit replace -p "/ARTIC/medaka MN908947.3" -r ""` 
-->
<section id="missing-intervals" class="level3" data-number="9.5.1">
<h3 data-number="9.5.1" class="anchored" data-anchor-id="missing-intervals"><span class="header-section-number">9.5.1</span> Missing Intervals</h3>
<p>As a further quality check, we also generated a table of missing intervals (indicated by the <code>N</code> character in the FASTA sequences). We used the <code>seqkit</code> software to achieve this.</p>
<p>First, we activate our software environment:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate seqkit</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Then, we ran the script <code>bash scripts/03-missing_intervals.sh</code>, which includes the following command:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> locate <span class="at">-i</span> <span class="at">-P</span> <span class="at">-G</span> <span class="at">-M</span> <span class="at">-r</span> <span class="at">-p</span> <span class="st">"N+"</span> report/consensus.fa <span class="op">&gt;</span> results/missing_intervals.tsv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This software outputs a tab-delimited table, which we saved as <code>results/missing_intervals.tsv</code>. The table looks like this (only the top few rows are shown):</p>
<pre><code>seqID  patternName  pattern  strand  start  end
ZA01   N+           N+       +       1      54
ZA01   N+           N+       +       22771  22926
ZA01   N+           N+       +       23603  23835
ZA01   N+           N+       +       26948  26948
ZA01   N+           N+       +       26968  27137
ZA01   N+           N+       +       29801  29867
ZA02   N+           N+       +       1      54
ZA02   N+           N+       +       22771  22921
ZA02   N+           N+       +       23603  23835</code></pre>
<p>We opened this file <code>missing_intervals.tsv</code> in <em>Excel</em> and quickly calculated the length of each interval. We noted that two samples - <em>ZA10</em> and <em>ZA23</em> - both have a continuous interval of over 18kb missing bases, which is not surprising as we had already identified these samples has having &gt;90% missing data. We make a note of these samples as being possibly problematic in downstream analysis steps.</p>
</section>
</section>
<section id="downstream-analyses" class="level2" data-number="9.6">
<h2 data-number="9.6" class="anchored" data-anchor-id="downstream-analyses"><span class="header-section-number">9.6</span> Downstream Analyses</h2>
<p>Based on the clean consensus sequences, we then perform several downstream analysis.</p>
<section id="lineage-assignment" class="level3" data-number="9.6.1">
<h3 data-number="9.6.1" class="anchored" data-anchor-id="lineage-assignment"><span class="header-section-number">9.6.1</span> Lineage Assignment</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>See <a href="../02-isolates/03-lineages.html#sec-lineages"><span>Section&nbsp;6.1</span></a>, if you need to revise how lineage assignment works.</p>
</div>
</div>
<p>Although the <em>Viralrecon</em> pipeline runs <em>Pangolin</em> and <em>Nextclade</em>, it does not use the latest version of these programs (because lineages evolve so fast, the nomenclature constantly changes). An up-to-date run of both of these tools can be done using each of their web applications:</p>
<ul>
<li><a href="https://clades.nextstrain.org/">clades.nextstrain.org</a></li>
<li><a href="https://pangolin.cog-uk.io/">pangolin.cog-uk.io</a></li>
</ul>
<p>However, for <strong>automation</strong>, <strong>reproducibility</strong> and <strong>traceability</strong> purposes, we used the command line versions of these tools, and included their analysis in two scripts.</p>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">Nextclade</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Pangolin</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<p>For <em>Nextclade</em>, we first activate the software environment:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate nextclade</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>And then we ran the script <code>bash scripts/04-nextclade.sh</code>, which contains the following commands:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get nextclade data</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="ex">nextclade</span> dataset get <span class="at">--name</span> sars-cov-2 <span class="at">--output-dir</span> resources/nextclade_background_data</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># run nextclade</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="ex">nextclade</span> run <span class="at">--input-dataset</span> resources/nextclade_background_data/ <span class="at">--output-all</span> results/nextclade report/consensus.fa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The first command downloads the latest version of the <em>Nextclade</em> background data using <code>nextclade dataset</code>. We use that data as input to the second command (<code>nextclade run</code>) to make sure it runs with the most up-to-date lineages.</p>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<p>For <em>Pangolin</em>, we first activate the software environment:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate pangolin</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>And then we ran the script <code>bash/04-pangolin.sh</code>, which contains the following commands:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># update pangolin data</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pangolin</span> <span class="at">--update-data</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># run pangolin</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="ex">pangolin</span> <span class="at">--outdir</span> results/pangolin/ <span class="at">--outfile</span> pango_report.csv report/consensus.fa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Similarly to before, we first ran <code>pangolin --update-data</code> to ensure we were using the latest lineages available. We can check the version of the data used with <code>pangolin --all-versions</code> (at the time we ran this we had <code>pangolin-data: 1.23.1</code>).</p>
</div>
</div>
</div>
<p>Both of these tools output CSV files, which can be open in <em>Excel</em> for further examination.<br>
Opening the <em>pangolin</em> results (<code>results/pangolin/pango_report.csv</code>), we noticed that the two problematic samples – ZA10 and ZA23 – failed the QC due to high fraction of missing data. The other samples all seemed to have been assigned to “Omicron” variant with a high support.</p>
<p>Opening the <em>nextclade</em> results (<code>results/nextclade/nextclade.tsv</code>), we noticed that the two problematic samples were classified as “recombinant”! We know from our quality control that we should not trust this assessment, and that most likely these are bad quality samples, not true recombinant lineages. Nextclade is more relaxed in assigning samples to lineages, so we should always check the QC status as well. We will notice that both of these samples were assigned QC status “bad”, due to their high percentage of missing data (<em>nextclade</em> uses a stringent threshold of 3000 sites, or ~10%, missing data). Two other samples had “bad” QC status. Sample ZA14 due to high % of missing data, and sample ZA18 from a mixture of a high number of private mutations and the presence of a frameshift mutation in ORF1b.</p>
<p>Like before, we will do further analysis (and visualisation) of these data using the software <em>R</em>, in the section “Integration &amp; Visualisation”, detailed below.</p>
</section>
<section id="phylogeny" class="level3" data-number="9.6.2">
<h3 data-number="9.6.2" class="anchored" data-anchor-id="phylogeny"><span class="header-section-number">9.6.2</span> Phylogeny</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>See <a href="../02-isolates/04-phylogeny.html#sec-phylogeny"><span>Section&nbsp;7.1</span></a>, if you need to revise how to build phylogenetic trees.</p>
</div>
</div>
<p>Although a tool such as <em>Nextclade</em> can place our samples in a global phylogeny context, sometimes it may be convient to build our own phylogenies. This requires three steps:</p>
<ul>
<li>Producing a multiple sequence alignment from all consensus sequences.</li>
<li>Tree inference.</li>
<li>Tree visualisation and annotation.</li>
</ul>
<p>Before our analysis, we first activated our software environment:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate phylo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>We performed the first two steps with the following script, which we ran with <code>bash scripts/05-phylogeny.sh</code>:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># alignment</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> results/mafft</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="ex">mafft</span> <span class="at">--6merpair</span> <span class="at">--maxambiguous</span> 0.2 <span class="at">--addfragments</span> report/consensus.fa resources/reference/sarscov2.fa <span class="op">&gt;</span> results/mafft/alignment.fa</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co"># tree inference</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> results/iqtree</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="ex">iqtree2</span> <span class="at">-s</span> results/mafft/alignment.fa <span class="at">--prefix</span> results/iqtree/consensus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The output of <em>iqtree</em> includes a tree file, which can be visualised using FigTree (or online using <a href="https://microreact.org/upload">Microreact</a>). The figure below shows our tree, which shows all our samples fall mostly clustering together. This makes sense, as all our samples were classified as “Omicron (BA.1-related)”.</p>
<p>It is worth noting that the samples ZA10, ZA14 and ZA23 are not included in this phylogeny, as they contained &gt;20% missing data (we used that threshold with MAFFT alignment, option <code>--maxambiguous 0.2</code>). Also, note that sample ZA18, which Nextclade had identified as “bad” QC (due to excess private mutations) also appears slightly separated from the other samples in the tree (the sample in red on the tree). The sample still clusters well with the rest, suggesting we can probably trust that it is indeed an Omicron variant, however the excess of private mutations (which may be due to sequencing errors) is making it stand apart from the others in the phylogeny.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/southafrica_phylogeny.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Phylogenetic tree for our samples. All samples were classified as lineage BA.1 (and some of its sub-lineages), corresponding to the Omicron VOC. The tips of the tree are coloured according to <em>Nextclade</em>’s QC status: green = “good”; blue = “mediocre”; red = “bad”.</figcaption>
</figure>
</div>
</section>
<section id="clustering" class="level3" data-number="9.6.3">
<h3 data-number="9.6.3" class="anchored" data-anchor-id="clustering"><span class="header-section-number">9.6.3</span> Clustering</h3>
<p>We identified groups of similar sequences in our data using the software <em>civet</em> (Cluster Investigation and Virus Epidemiology Tool). This software compares our samples with a background dataset of our choice, which givus us more context for our analysis. In this case we are using the <a href="https://github.com/artic-network/civet/tree/master/quickstart_data/civet_background_data">example background data</a> that comes with <em>civet</em>. However, in a real-world analysis, it would have been ideal to choose local samples as background data. For example, we could download samples from South Africa from around the time period of our sample collection, from GISAID following the instructions on the <a href="https://cov-lineages.org/resources/civet/walkthrough.html#background_dataset"><em>civet</em> documentation</a> (you need an account on GISAID to obtain these data).</p>
<p>For this example, we already prepared <em>civet</em> background dataset saved in <code>resources/civet_background_data</code>.</p>
<p>Before our analysis, we first activate our software environment:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate civet</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Then, we ran the script <code>bash scripts/06-civet.sh</code>, which contains the following code:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run civet analysis</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="ex">civet</span> <span class="dt">\</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-i</span> sample_info.csv <span class="dt">\</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">-f</span> report/consensus.fa <span class="dt">\</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">-icol</span> sample <span class="dt">\</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">-idate</span> collection_date <span class="dt">\</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-d</span> resources/civet_background_data/ <span class="dt">\</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">-o</span> results/civet</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The result of this analysis includes an interactive HTML report (in <code>results/civet/civet.html</code>). We can see that our samples were grouped into a single catchment. This makes sense from our previous lineage/variant analysis: all our samples were classfied as Omicron VOC. We can see that our samples seem to be more diverged from the Australian BA.1 sample present in the background data, with new SNPs in our samples creating a longer branch in the tree.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/southafrica_civet.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Results from <em>civet</em> report for catchment 1, showing the phylogeny in the context of a BA.1 Australian sample from the background data.</figcaption>
</figure>
</div>
<p><em>Civet</em> also outputs a CSV file (<code>results/civet/master_metadata.csv</code>), which includes the catchment that each sample was assigned to. We will use this CSV file later to integrate this information with other parts of our analysis, in <em>R</em>, detailed in the “Integration &amp; Visualisation” section.</p>
</section>
</section>
<section id="integration-visualisation" class="level2" data-number="9.7">
<h2 data-number="9.7" class="anchored" data-anchor-id="integration-visualisation"><span class="header-section-number">9.7</span> Integration &amp; Visualisation</h2>
<p>At this point in our analysis, we have several tables with different pieces of information:</p>
<ul>
<li><code>sample_info.csv</code> → the original table with metadata for our samples.</li>
<li><code>results/viralrecon/multiqc/medaka/summary_variants_metrics_mqc.csv</code> → quality metrics from the <em>MultiQC</em> report generated by the <em>viralrecon</em> pipeline.</li>
<li><code>results/nextclade/nextclade.tsv</code> → the results from <em>Nextclade</em>.</li>
<li><code>results/pangolin/pango_report.csv</code> → the results from <em>Pangolin</em>.</li>
<li><code>results/civet/master_metadata.csv</code> → the results from the <em>civet</em> analysis, namely the catchment (or cluster) that each of our samples was grouped into.</li>
</ul>
<p>To consolidate our analysis, we <strong>tidied and integrated</strong> the information from across these different files, into a single table using the software <em>R</em>. The script used to do this is in <code>scripts/07-data_integration.R</code>. Because this is an <em>R</em> script, we opened it in <em>RStudio</em> to execute the code.</p>
<p>The output of our script is a new tab-delimited table, which we saved in <code>report/consensus_metrics.tsv</code>, and contains the following columns:</p>
<ul>
<li><code>sample</code> → sample ID.</li>
<li><code>collection_date</code> → date of collection day.</li>
<li><code>collection_week</code> → date of collection week (useful for summarising/visualising counts per-week).</li>
<li><code>country</code> → country of origin.</li>
<li><code>latitude</code>/<code>longitude</code> → latitude and longitude of collection.</li>
<li><code>n_mapped_reads</code> → number of mapped reads.</li>
<li><code>median_depth</code> → median depth.</li>
<li><code>pct_missing</code> → percentage of missing data.</li>
<li><code>pct_coverage</code> → percentage of coverage.</li>
<li><code>n_variants</code> → number of SNP + indel variants detected.</li>
<li><code>nextclade</code> → nextclade clade.</li>
<li><code>qc_status</code> → QC status as determined by <em>Nextclade</em> (“bad”, “mediocre”, “good”).</li>
<li><code>lineage</code> → <em>Pangolin</em> lineage.</li>
<li><code>who_variant</code> → variant of concern designation.</li>
<li><code>catchment</code> → catchment group from <em>Civet</em>.</li>
</ul>
<p>This table, which aggregates information from many of the tools we used, was then used to produce different visualisations of our analysis. These visualisations were also done using the <em>R</em> software (<code>scripts/08-visualisation.R</code>), and integrated into a report, shown below.</p>
<iframe src="https://docs.google.com/document/d/e/2PACX-1vSbIIVjqtDFnfEbw5tQzPY1z24FLWQHUaDvFV6_bNqVvwd1d6gEZQ1djJrMTEd5MmQOU8pwA5Q-MuRg/pub?embedded=true" width="100%" height="500">
</iframe>
</section>
<section id="bonus-full-workflow" class="level2" data-number="9.8">
<h2 data-number="9.8" class="anchored" data-anchor-id="bonus-full-workflow"><span class="header-section-number">9.8</span> Bonus: Full Workflow</h2>
<p>Although we have ran each of the steps of our analysis individually (each in their own script), now that we have everything working, we could integrate all these steps into a single “master” script:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co"># make mamba activate command available</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="bu">eval</span> <span class="st">"</span><span class="va">$(</span><span class="ex">conda</span> shell.bash hook<span class="va">)</span><span class="st">"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> <span class="va">$(</span><span class="ex">mamba</span> info <span class="at">--base</span><span class="va">)</span>/etc/profile.d/mamba.sh</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co"># make report directory</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> report</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate nextflow</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="co"># run viralrecon</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="ex">nextflow</span> run nf-core/viralrecon <span class="dt">\</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">-r</span> 2.6.0 <span class="at">-profile</span> singularity <span class="dt">\</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">--max_memory</span> <span class="st">'15.GB'</span> <span class="at">--max_cpus</span> 8 <span class="dt">\</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">--platform</span> illumina <span class="dt">\</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input</span> samplesheet.csv <span class="dt">\</span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">--outdir</span> results/viralrecon <span class="dt">\</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">--protocol</span> amplicon <span class="dt">\</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">--genome</span> <span class="st">'MN908947.3'</span> <span class="dt">\</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">--primer_set</span> artic <span class="dt">\</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">--primer_set_version</span> 3 <span class="dt">\</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">--skip_assembly</span> <span class="at">--skip_asciigenome</span> <span class="dt">\</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">--skip_pangolin</span> <span class="at">--skip_nextclade</span></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="co"># combine and clean FASTA files</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> results/viralrecon/variants/ivar/consensus/bcftools/<span class="pp">*</span>.consensus.fa <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s| MN908947.3||'</span> <span class="op">&gt;</span> report/consensus.fa</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate seqkit</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a><span class="co"># create missing bases TSV file</span></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> locate <span class="at">-i</span> <span class="at">-P</span> <span class="at">-G</span> <span class="at">-M</span> <span class="at">-r</span> <span class="at">-p</span> <span class="st">"N+"</span> report/consensus.fa <span class="op">&gt;</span> results/missing_intervals.tsv</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate nextclade</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a><span class="co"># get nextclade data</span></span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a><span class="ex">nextclade</span> dataset get <span class="at">--name</span> sars-cov-2 <span class="at">--output-dir</span> resources/nextclade_background_data</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a><span class="co"># run nextclade</span></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a><span class="ex">nextclade</span> run <span class="at">--input-dataset</span> resources/nextclade_background_data/ <span class="at">--output-all</span> results/nextclade report/consensus.fa</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate pangolin</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a><span class="co"># update pangolin data</span></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a><span class="ex">pangolin</span> <span class="at">--update-data</span></span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a><span class="co"># run pangolin</span></span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a><span class="ex">pangolin</span> <span class="at">--outdir</span> results/pangolin/ <span class="at">--outfile</span> pango_report.csv report/consensus.fa</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate phylo</span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a><span class="co"># alignment</span></span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> results/mafft</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a><span class="ex">mafft</span> <span class="at">--6merpair</span> <span class="at">--maxambiguous</span> 0.2 <span class="at">--addfragments</span> report/consensus.fa resources/reference/sarscov2.fa <span class="op">&gt;</span> results/mafft/alignment.fa</span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a><span class="co"># tree inference</span></span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> results/iqtree</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a><span class="ex">iqtree2</span> <span class="at">-s</span> results/mafft/alignment.fa <span class="at">--prefix</span> results/iqtree/consensus</span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a><span class="co"># data integration and cleaning</span></span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> scripts/07-data_integration.R</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Notice that we included the <em>R</em> script that does the data cleaning here, using the <code>Rscript</code> program that allows to execute an <em>R</em> script from the command-line.</p>
<p>Having this “master” script, we could run all these steps from start-to-finish with a single command, which can be very useful if you want to fully automate your analysis across multiple runs.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../materials/03-case_studies/01-switzerland.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Switzerland (Nanopore)</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../materials/03-case_studies/03-eqa.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">EQA (Exercise)</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">Licensed <a href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a> <img src="https://creativecommons.org/images/deed/cc_icon_black_x2.png" class="img-fluid" width="25"> Bioinformatics Training Facility</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>